### DISCUSSION TITLE
[ISSUE] - setting up dev environment

### DISCUSSION THREAD

**Discussion by Tryd3x** [online-ml/river#1450] [Q&A]
I would like to get my hands dirty on my first contribution on GitHub in areas related to Machine Learning. I came across an issue [issue#1399](https://github.com/online-ml/river/issues/1399) which piqued my interest and as a newbie would like to start working on it. I would appreciate what are first steps I need to take in order to get the just of it.

Progress:
I've finished setting up the development environment as mentioned in [Contribution.md](https://github.com/online-ml/river/blob/main/CONTRIBUTING.md). Steps I've taken:

1. Using VScode IDE, forked the river repository and cloned it into local environment (Windows 10 with mingw64 for bash shell)
2. Using pyenv-win since I'm using windows (can't run a LinuxVM and conduct development)
3. Installed poetry for dependency management.
4. Configured pre-commit push hooks.

so far I have a running dev environment but I'm not sure if I configured it the right way as mentioned in the [contribution.md](https://github.com/online-ml/river/blob/main/CONTRIBUTING.md) and would like to know the ideal way of setting up the development environment.

[**_Apologies in advance for being a newbie and not following a proper format for starting a discussion]_**

**Comment by MaxHalford** [comment]
Hey there.

> so far I have a running dev environment but I'm not sure if I configured it the right way as mentioned in the [contribution.md](https://github.com/online-ml/river/blob/main/CONTRIBUTING.md) and would like to know the ideal way of setting up the development environment.

I'm not sure what your question? If you've followed the steps, what is the issue?

Maybe one you can do is run `pytest` in the terminal. If it runs without errors, then your environment is ready.

**Comment by Tryd3x** [comment]
Apologies for not stating the purpose of the thread clearly. 

I'm not reporting any issues of such. I was just posting this thread to receive opinions/advice from people regarding the way I have set up my development environment (Generally people opt developing on Linux or Ubuntu based OS as their dev environment). I was wondering if using Windows is the way to go or are there other ways one would've set their environment up differently (Which is the preferred approach?)

**TLDR: Not an ISSUE, more of a DISCUSSION thread**

> (once again, I'm a newbie :d. If this is posted in the wrong thread, you can proceed to move it somewhere else)

**Reply by MaxHalford** [reply]
You can Windows yes :). As long you're comfortable with your setup, you're good to go

**Comment by Tryd3x** [comment]
## **Issue**
- river(0.20.1) failed to install

## **Steps executed that caused issue**
1. Forked repository online-ml/river
2. Cloned repository to local system (Windows 10 with mingw64 for bash shell)
3. Installed poetry (1.7.0)
4. Installed pyenv(3.1.1) (I'm using windows (can't run a LinuxVM and conduct development)
5. Navigated to project path and ran the command: 
> poetry shell
> poetry install
6. Error pops up after all dependencies are installed.

![image](https://github.com/online-ml/river/assets/43925824/32558554-1eaa-405a-91bf-fc67b273e371)


Also received this warning mid installation of dependencies:
![image](https://github.com/online-ml/river/assets/43925824/e3c219e1-894b-4e07-b5e6-85128a865d67)

**Reply by MaxHalford** [reply]
I have no idea. It seems like you're using Python 2?

**Reply by Tryd3x** [reply]
I'm using python 3.10

**Reply by MaxHalford** [reply]
I'm sorry but I don't have a Windows computer so I'm not able to reproduce. It is an issue though.

**Comment by Tryd3x** [comment]
# **RESOLVED**
I missed out on installing rust. :p
